File: memory-api/src/main/java/com/memory/service/admin/MemberAdminService.java
package com.memory.service.admin;

import com.memory.config.jwt.JwtTokenProvider;
import com.memory.domain.member.Member;
import com.memory.domain.member.MemberType;
import com.memory.domain.member.repository.MemberRepository;
import com.memory.dto.member.MemberRequest;
import com.memory.dto.member.response.MemberLoginResponse;
import com.memory.exception.customException.NotFoundException;
import com.memory.exception.customException.ValidationException;
import lombok.RequiredArgsConstructor;
import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
@RequiredArgsConstructor
public class MemberAdminService {

    private final MemberRepository memberRepository;
    private final PasswordEncoder passwordEncoder;
    private final JwtTokenProvider jwtTokenProvider;

    @Transactional(readOnly = true)
    public MemberLoginResponse adminLogin(MemberRequest.Login loginRequestDto) {
        Member member = memberRepository.findMemberByEmailAndMemberType(loginRequestDto.getEmail(), MemberType.ADMIN)
                .orElseThrow(() -> new NotFoundException("존재하지 않는 관리자 이메일입니다."));

        if (!passwordEncoder.matches(loginRequestDto.getPassword(), member.getPassword())) {
            throw new ValidationException("비밀번호가 일치하지 않습니다.");
        }

        String accessToken = jwtTokenProvider.createAccessToken(member.getEmail());
        String refreshToken = jwtTokenProvider.createRefreshToken(member.getEmail());
        
        return MemberLoginResponse.of(accessToken, refreshToken);
    }
}

File: memory-api/src/main/kotlin/com/memory/service/admin/MemberAdminService.kt
package com.memory.service.admin

import com.memory.config.jwt.JwtTokenProvider
import com.memory.domain.member.MemberType
import com.memory.domain.member.repository.MemberRepository
import com.memory.dto.member.MemberRequest
import com.memory.dto.member.response.MemberLoginResponse
import com.memory.exception.customException.NotFoundException
import com.memory.exception.customException.ValidationException
import org.springframework.security.crypto.password.PasswordEncoder
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
class MemberAdminService(
    private val memberRepository: MemberRepository,
    private val passwordEncoder: PasswordEncoder,
    private val jwtTokenProvider: JwtTokenProvider,
) {

    @Transactional(readOnly = true)
    fun adminLogin(request: MemberRequest.Login): MemberLoginResponse {
        val member = memberRepository.findMemberByEmailAndMemberType(request.email, MemberType.ADMIN)
            .orElseThrow { NotFoundException("존재하지 않는 회원입니다.") }

        if (!passwordEncoder.matches(request.password, member.password)) {
            throw ValidationException("비밀번호가 일치하지 않습니다.")
        }
        val accessToken = jwtTokenProvider.createAccessToken(member.email)
        val refreshToken = jwtTokenProvider.createRefreshToken(member.email)

        return MemberLoginResponse.of(accessToken, refreshToken)
    }
}